src/Splunk.OpenTelemetry.AutoInstrumentation/Helpers/ServiceNameHelper.cs

```diff
@@ -0,0 +1,98 @@
+﻿// <copyright file="ServiceNameHelper.cs" company="Splunk Inc.">
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Splunk.OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace Splunk.OpenTelemetry.AutoInstrumentation.Helpers
+{
+    internal class ServiceNameHelper
+    {
+        private static readonly ILogger Log = new Logger();
+
+        public static bool HasServiceName(PluginSettings settings)
+        {
+            return !string.IsNullOrEmpty(settings.ServiceName) || ResourceHasServiceName(settings.ResourceAttributes);
+
+            bool ResourceHasServiceName(IEnumerable<KeyValuePair<string, string>> resources)
+            {
+                const string key = "service.name";
+
+                var serviceName = settings.ResourceAttributes
+                    .FirstOrDefault(x => x.Key == key)
+                    .Value;
+
+                return !string.IsNullOrEmpty(serviceName);
+            }
+        }
+
+        public static string GetGeneratedServiceName()
+        {
+            try
+            {
+#if NETFRAMEWORK
+                try
+                {
+                    if (TryLoadAspNetSiteName(out var siteName))
+                    {
+                        return siteName!;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // Unable to call into System.Web.dll
+                    Log.Error(ex, "Unable to get application name through ASP.NET settings");
+                }
+#endif
+
+                var serviceName = Assembly.GetEntryAssembly()?.GetName().Name ??
```

Comment:
So for `dotnet run testApplicatiom" the name should be `testApplication`?

Reply:
I can't make dotnet run to work at all 😞 Seems similar issue like `dotnet help`, as we didn't actually fix it.
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1744

Here is the screen for running as exe
![image](https://user-images.githubusercontent.com/4929112/207027660-422e9b07-a3ad-4271-be79-677b26861472.png)

Reply:
Seems to get me to the next point but unfortunately does not solve it.
We should really consider baking it into our solution, if it doesn't work with dotnet.exe.
`$env:OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES = "dotnet,dotnet.exe"`

I'm currently not sure why the profiler still engages.
![image](https://user-images.githubusercontent.com/4929112/207095077-b3b7367c-03c4-4e69-b74f-ddf9008476e7.png)

Reply:
Are we missing some dependency on the AdditionalDeps? Do you get the same error just with the DOTNET_* env vars?

Reply:
AdditionalDeps is not supported for self-contained apps, see https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1699#issuecomment-1334204352

It is pretty inconvenient in this case because if you set the env vars then basically you can't use `dotnet` CLI ... 🤔

Reply:
Somehow works now, when I switched to VS Code PowerShell terminal 🤔 
`$env:OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES = "dotnet,dotnet.exe"`

![image](https://user-images.githubusercontent.com/4929112/207131275-3ffff295-7c85-43d7-ad49-b48b104e5c2c.png)

Reply:
How did you launch the first terminal? Was it a VS developer PS session?

Reply:
no, just a normal PowerShell session.